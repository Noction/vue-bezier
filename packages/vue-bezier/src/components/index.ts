// Exporting components from Collapse directory
export { default as CollapseTransition } from './Collapse/CollapseTransition.vue';

// Exporting components from Fade directory
export { default as FadeTransition } from './Fade/FadeTransition.vue';

// Exporting components from Scale directory
export { default as ScaleTransition } from './Scale/ScaleTransition.vue';

// Exporting components from Slide directory
export { default as SlideXLeftTransition } from './Slide/SlideXLeftTransition.vue';
export { default as SlideXRightTransition } from './Slide/SlideXRightTransition.vue';
export { default as SlideYDownTransition } from './Slide/SlideYDownTransition.vue';
export { default as SlideYUpTransition } from './Slide/SlideYUpTransition.vue';

// Exporting components from Zoom directory
export { default as ZoomCenterTransition } from './Zoom/ZoomCenterTransition.vue';
export { default as ZoomUpTransition } from './Zoom/ZoomUpTransition.vue';
export { default as ZoomXTransition } from './Zoom/ZoomXTransition.vue';
export { default as ZoomYTransition } from './Zoom/ZoomYTransition.vue';
