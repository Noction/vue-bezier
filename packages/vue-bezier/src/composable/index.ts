import buildComponentType from './buildComponentType'
import buildEmits from './buildEmits'
import buildHooks from './buildHooks'
import buildProps from './buildProps'
import buildTag from './buildTag'

export { buildComponentType, buildEmits, buildHooks, buildProps, buildTag }
