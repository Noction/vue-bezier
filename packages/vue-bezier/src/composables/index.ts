import buildComponentType from './buildComponentType'
import useHooks from './useHooks'

export { buildComponentType, useHooks }
