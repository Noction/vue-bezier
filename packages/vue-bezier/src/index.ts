import PrivateCollapseTransition from './components/Collapse/CollapseTransition.vue'

import PrivateFadeTransition from './components/Fade/FadeTransition.vue'

import PrivateScaleTransition from './components/Scale/ScaleTransition.vue'

import PrivateSlideXLeftTransition from './components/Slide/SlideXLeftTransition.vue'
import PrivateSlideXRightTransition from './components/Slide/SlideXRightTransition.vue'
import PrivateSlideYDownTransition from './components/Slide/SlideYDownTransition.vue'
import PrivateSlideYUpTransition from './components/Slide/SlideYUpTransition.vue'

import PrivateZoomCenterTransition from './components/Zoom/ZoomCenterTransition.vue'
import PrivateZoomUpTransition from './components/Zoom/ZoomUpTransition.vue'
import PrivateZoomXTransition from './components/Zoom/ZoomXTransition.vue'
import PrivateZoomYTransition from './components/Zoom/ZoomYTransition.vue'

const components = {
  CollapseTransition: PrivateCollapseTransition,
  FadeTransition: PrivateFadeTransition,
  ScaleTransition: PrivateScaleTransition,
  SlideXLeftTransition: PrivateSlideXLeftTransition,
  SlideXRightTransition: PrivateSlideXRightTransition,
  SlideYDownTransition: PrivateSlideYDownTransition,
  SlideYUpTransition: PrivateSlideYUpTransition,
  ZoomCenterTransition: PrivateZoomCenterTransition,
  ZoomUpTransition: PrivateZoomUpTransition,
  ZoomXTransition: PrivateZoomXTransition,
  ZoomYTransition: PrivateZoomYTransition
}

export {
  PrivateCollapseTransition as CollapseTransition,
  PrivateFadeTransition as FadeTransition,
  PrivateScaleTransition as ScaleTransition,
  PrivateSlideXLeftTransition as SlideXLeftTransition,
  PrivateSlideXRightTransition as SlideXRightTransition,
  PrivateSlideYDownTransition as SlideYDownTransition,
  PrivateSlideYUpTransition as SlideYUpTransition,
  PrivateZoomCenterTransition as ZoomCenterTransition,
  PrivateZoomUpTransition as ZoomUpTransition,
  PrivateZoomXTransition as ZoomXTransition,
  PrivateZoomYTransition as ZoomYTransition
}

export function install (app) {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  if (app.$_v3SFCTransitionsInstalled) return
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  app.$_v3SFCTransitionsInstalled = true

  Object
    .entries(components)
    .forEach(([key, component]) =>
      app.component(key, component)
    )
}

export default {
  install
}
