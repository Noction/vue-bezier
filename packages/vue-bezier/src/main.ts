import { App } from 'vue'

import CollapseTransition from './components/Collapse/CollapseTransition.vue'

import FadeTransition from './components/Fade/FadeTransition.vue'

import ScaleTransition from './components/Scale/ScaleTransition.vue'

import SlideXLeftTransition from './components/Slide/SlideXLeftTransition.vue'
import SlideXRightTransition from './components/Slide/SlideXRightTransition.vue'
import SlideYDownTransition from './components/Slide/SlideYDownTransition.vue'
import SlideYUpTransition from './components/Slide/SlideYUpTransition.vue'

import ZoomCenterTransition from './components/Zoom/ZoomCenterTransition.vue'
import ZoomUpTransition from './components/Zoom/ZoomUpTransition.vue'
import ZoomXTransition from './components/Zoom/ZoomXTransition.vue'
import ZoomYTransition from './components/Zoom/ZoomYTransition.vue'

export const components = {
  CollapseTransition,
  FadeTransition,
  ScaleTransition,
  SlideXLeftTransition,
  SlideXRightTransition,
  SlideYDownTransition,
  SlideYUpTransition,
  ZoomCenterTransition,
  ZoomUpTransition,
  ZoomXTransition,
  ZoomYTransition
}

export function install (app: App) {
  // @ts-expect-error it's a custom flag
  if (app.$_vueBezierInstalled) return
  // @ts-expect-error it's a custom flag
  app.$_vueBezierInstalled = true

  Object
    .entries(components)
    .forEach(([key, component]) =>
      app.component(key, component)
    )
}

export default {
  install
}
