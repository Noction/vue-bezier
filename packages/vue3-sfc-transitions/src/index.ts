import PrivateCollapseTransition from './components/Collapse/CollapseTransition.vue'

import PrivateFadeTransition from './components/Fade/FadeTransition.vue'

import PrivateScaleTransition from './components/Scale/ScaleTransition.vue'

import PrivateSlideXLeftTransition from './components/Slide/SlideXLeftTransition.vue'
import PrivateSlideXRightTransition from './components/Slide/SlideXRightTransition.vue'
import PrivateSlideYDownTransition from './components/Slide/SlideYDownTransition.vue'
import PrivateSlideYUpTransition from './components/Slide/SlideYUpTransition.vue'

import PrivateZoomCenterTransition from './components/Zoom/ZoomCenterTransition.vue'
import PrivateZoomUpTransition from './components/Zoom/ZoomUpTransition.vue'
import PrivateZoomXTransition from './components/Zoom/ZoomXTransition.vue'
import PrivateZoomYTransition from './components/Zoom/ZoomYTransition.vue'

interface Options {
  casing?: 'PascalCase' | 'kebab-case'
  components?: Array<
    | 'FadeTransition'
    | 'ZoomCenterTransition'
    | 'ZoomXTransition'
    | 'ZoomYTransition'
    | 'ZoomUpTransition'
    | 'SlideYUpTransition'
    | 'SlideYDownTransition'
    | 'SlideXLeftTransition'
    | 'SlideXRightTransition'
    | 'ScaleTransition'
    | 'CollapseTransition'
  >
}

const components = {
  CollapseTransition: PrivateCollapseTransition,
  FadeTransition: PrivateFadeTransition,
  ScaleTransition: PrivateScaleTransition,
  SlideXLeftTransition: PrivateSlideXLeftTransition,
  SlideXRightTransition: PrivateSlideXRightTransition,
  SlideYDownTransition: PrivateSlideYDownTransition,
  SlideYUpTransition: PrivateSlideYUpTransition,
  ZoomCenterTransition: PrivateZoomCenterTransition,
  ZoomUpTransition: PrivateZoomUpTransition,
  ZoomXTransition: PrivateZoomXTransition,
  ZoomYTransition: PrivateZoomYTransition
}

export {
  PrivateCollapseTransition as CollapseTransition,
  PrivateFadeTransition as FadeTransition,
  PrivateScaleTransition as ScaleTransition,
  PrivateSlideXLeftTransition as SlideXLeftTransition,
  PrivateSlideXRightTransition as SlideXRightTransition,
  PrivateSlideYDownTransition as SlideYDownTransition,
  PrivateSlideYUpTransition as SlideYUpTransition,
  PrivateZoomCenterTransition as ZoomCenterTransition,
  PrivateZoomUpTransition as ZoomUpTransition,
  PrivateZoomXTransition as ZoomXTransition,
  PrivateZoomYTransition as ZoomYTransition
}

export function install (app, options?: Options) {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  if (app.$_v3SFCTransitionsInstalled) return
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  app.$_v3SFCTransitionsInstalled = true

  const componentNameCasing = {
    PascalCase: str => str,
    'kebab-case': str =>
      `${str.slice(0, 1)}${str
        .slice(1)
        .replace(/[A-Z]/g, '-$&')}`.toLocaleLowerCase()
  }

  const nameFormat = componentNameCasing[options?.casing ?? 'kebab-case']

  if (options && options?.components) {
    options.components.forEach((name: string) =>
      app.component(nameFormat(name), components[name])
    )
  } else {
    Object.entries(components).forEach(([key, component]) =>
      app.component(nameFormat(key), component)
    )
  }
}

export default {
  install
}
