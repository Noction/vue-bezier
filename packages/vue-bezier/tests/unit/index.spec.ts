import { expect } from 'vitest'
import { faker } from '@faker-js/faker'
import { install } from '../../src'
import { createApp, defineComponent } from 'vue'

describe('index', () => {
  it('When installing with default argument, all of the transition component will be registered globally', () => {
    const rootComponent = defineComponent({ name: faker.lorem.word() })
    const appInstance = createApp(rootComponent)

    install(appInstance)

    expect(appInstance._context.components).toMatchSnapshot({
      CollapseTransition: expect.any(Object),
      FadeTransition: expect.any(Object),
      ScaleTransition: expect.any(Object),
      SlideXLeftTransition: expect.any(Object),
      SlideXRightTransition: expect.any(Object),
      SlideYDownTransition: expect.any(Object),
      SlideYUpTransition: expect.any(Object),
      ZoomCenterTransition: expect.any(Object),
      ZoomUpTransition: expect.any(Object),
      ZoomXTransition: expect.any(Object),
      ZoomYTransition: expect.any(Object)
    })
  })

  it('When installed flag is `true`, it should not install additional components', () => {
    const rootComponent = defineComponent({ name: faker.lorem.word() })
    const appInstance = createApp(rootComponent)

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    appInstance.$_v3SFCTransitionsInstalled = true

    install(appInstance)

    expect(appInstance._context.components).toStrictEqual({})
  })
})
