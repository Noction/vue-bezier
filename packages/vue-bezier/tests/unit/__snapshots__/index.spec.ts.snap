// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > When installing with default argument, all of the transition component will be registered globally 1`] = `
{
  "CollapseTransition": Any<Object>,
  "FadeTransition": Any<Object>,
  "ScaleTransition": Any<Object>,
  "SlideXLeftTransition": Any<Object>,
  "SlideXRightTransition": Any<Object>,
  "SlideYDownTransition": Any<Object>,
  "SlideYUpTransition": Any<Object>,
  "ZoomCenterTransition": Any<Object>,
  "ZoomUpTransition": Any<Object>,
  "ZoomXTransition": Any<Object>,
  "ZoomYTransition": Any<Object>,
}
`;
